Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs

```diff
@@ -103,5 +105,76 @@ public TestsRepository(TrappistDbContext dbContext, IGlobalUtil util)
             await _dbContext.SaveChangesAsync();
         }
         #endregion
+        #region Category selection
+        public async Task<TestAC> GetTestDetailsByIdAsync(int testId)
+        {
+            var testAcObject = new TestAC();
+            var test = await _dbContext.Test.FindAsync(testId);
+            testAcObject = Mapper.Map<Test, TestAC>(test);
+            var categoryList = await _dbContext.Category.ToListAsync();
+            var categoryListAc = Mapper.Map<List<Category>, List<CategoryAC>>(categoryList);
+            var testCategoryList = await _dbContext.TestCategory.Where(x => x.TestId == testId).Include(x => x.Category).ToListAsync();
+            categoryListAc.ForEach(category =>
+            {
+                if (testCategoryList.Exists(x => x.CategoryId == category.Id))
+                    category.IsSelect = true;
+            });
+            testAcObject.CategoryAcList = categoryListAc;
+            return testAcObject;
+        }
+
+        public async Task AddSelectedCategoryAsync(List<TestCategory> testCategory)
+        {
+            var testCategoryList = new List<TestCategory>();
+            var testCategories = await _dbContext.TestCategory.ToListAsync();
+            foreach (var category in testCategory)
+            {
+                if (!testCategories.Exists(x => x.CategoryId == category.CategoryId && x.TestId == category.TestId))
+                    testCategoryList.Add(category);
+            }
+            await _dbContext.TestCategory.AddRangeAsync(testCategoryList);
+            await _dbContext.SaveChangesAsync();
+        }
+
+        public async Task<bool> DeselectCategoryAync(int categoryId, int testId)
+        {
+            var questions = await _dbContext.Question.OrderBy(x => x.CategoryID).ToListAsync();
+            var testQuestions = await _dbContext.TestQuestion.OrderBy(x => x.QuestionId).ToListAsync();
+            foreach (var question in questions)
+            {
+                if (question.CategoryID == categoryId)
+                    if (testQuestions.Exists(testQuestion => testQuestion.QuestionId == question.Id && testQuestion.TestId == testId))
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1066)